models/note.go

```diff
@@ -0,0 +1,34 @@
+// Package models defines the data types, payloads and repository interfaces
+// of the accounts service.
+package models
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+)
+
+type NoteWithBlocks struct {
+	ID       uuid.UUID `json:"id" bson:"_id,omitempty"`
+	AuthorId string    `json:"authorId" bson:"authorId,omitempty"`
+	Title    string    `json:"title" bson:"title,omitempty"`
+	Blocks   []Block   `json:"blocks" bson:"blocks,omitempty"`
+}
+
+type NoteFilter struct {
+	ID       uuid.UUID `json:"id" bson:"_id,omitempty"`
+	AuthorId string    `json:"authorId" bson:"authorId,omitempty"`
+}
```

Comment:
Same thing as with the blocks. Using this filter type, you can either target one note by ID or a group of notes (given a user can be the author of multiple notes). I think it's better to have a `NoteFilter` and `ManyNotesFilter` and to differentiate `Delete`, `Get`, `Update` operations (which delete, get and update one single note) and `DeleteMany`, `List` and `UpdateMany` operations. I'm open to talk discuss over Discord if you want.